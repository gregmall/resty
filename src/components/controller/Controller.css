.url{
  width: 800px;
}
