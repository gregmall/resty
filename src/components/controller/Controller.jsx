import React from 'react';
import PropTypes from 'prop-types';
import styles from './Controller.css'

const Controller = ({
  url,
  text,
  onChange,
  onSubmit,
  
  

}) => (

  <form onSubmit={onSubmit} className={styles.Controller}>
  <label htmlFor="url">ENTER URL</label>
  <input
    className="url"
    id="url"
    type="url"
    name="url"
    value={url}
    onChange={onChange}
   
    /><br />
    <label htmlFor="method">GET</label>
    <input
    id="method"
    type="radio"
    name="method"
    value='GET'
    onChange={onChange}
    /><br />
     <label htmlFor="method">POST</label>
    <input
    id="method"
    type="radio"
    name="method"
    value='POST'
    onChange={onChange}
    /><br />
     <label htmlFor="method">DELETE</label>
    <input
    id="method"
    type="radio"
    name="method"
    value='DELETE'
    onChange={onChange}
    /><br />
     <label htmlFor="method">PUT</label>
    <input
    id="method"
    type="radio"
    name="method"
    value='PUT'
    onChange={onChange}
    /><br />
    <label htmlFor="text">enter json</label>
  <textarea
    className="text"
    id="text"
    type="textarea"
    name="text"
    value={text}
    onChange={onChange}
   
    /><br />
    
    <button type="submit">SUBMIT</button>

</form>

);
Controller.propTypes = {
  url: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};
export default Controller;
