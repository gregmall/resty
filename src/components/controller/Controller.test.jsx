import React from 'react';
import { render, cleanup } from '@testing-library/react';
import  Controller from './Controller';

describe('Controller component', () => {
  afterEach(() => cleanup());

  it('renders Controller', () => {
    const { asFragment } = render(<Controller
      url="www.url.com"
      text="Test Display"
      method="method"
      handleChange={() => {}}
      handleSubmit={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
