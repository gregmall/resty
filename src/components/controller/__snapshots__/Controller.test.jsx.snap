// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controller component renders Controller 1`] = `
<DocumentFragment>
  <form
    class="Controller"
  >
    <label
      for="url"
    >
      ENTER URL
    </label>
    <input
      class="url"
      id="url"
      name="url"
      type="url"
      value="www.url.com"
    />
    <br />
    <label
      for="method"
    >
      GET
    </label>
    <input
      id="method"
      name="method"
      type="radio"
      value="GET"
    />
    <br />
    <label
      for="method"
    >
      POST
    </label>
    <input
      id="method"
      name="method"
      type="radio"
      value="POST"
    />
    <br />
    <label
      for="method"
    >
      DELETE
    </label>
    <input
      id="method"
      name="method"
      type="radio"
      value="DELETE"
    />
    <br />
    <label
      for="method"
    >
      PUT
    </label>
    <input
      id="method"
      name="method"
      type="radio"
      value="PUT"
    />
    <br />
    <label
      for="text"
    >
      enter json
    </label>
    <textarea
      class="text"
      id="text"
      name="text"
      type="textarea"
    >
      Test Display
    </textarea>
    <br />
    <button
      type="submit"
    >
      SUBMIT
    </button>
  </form>
</DocumentFragment>
`;
